packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts

```diff
@@ -121,23 +121,23 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
     const { fieldNames } = metadataMap;
 
     if (!idField.value) {
-      throw new NoDataError(`Cannot delete ${this.collectionName} - missing ID field`)
+      throw new NoDataError(`Couldn't get document from ${this.collectionName} - missing ID field`)
     }
     const projection = {
       ...this.buildProjectionOption(context),
       [fieldNames.updatedAt]: 1,
       _deleted: 1
     };
-    const queryResult = await this.db.collection(this.collectionName).findOne({ _id: new ObjectId(idField.value), _deleted: false }, { projection });
+    const queryResult = await this.db.collection(this.collectionName).findOne({ _id: new ObjectId(idField.value), _deleted: { $ne: true} }, { projection });
     if (queryResult) {
       queryResult[idField.name] = queryResult._id;
-      if (clientData[fieldNames.updatedAt].toString() !== queryResult[fieldNames.updatedAt].toString()) {
+      if (queryResult[fieldNames.updatedAt] !== undefined && clientData[fieldNames.updatedAt].toString() !== queryResult[fieldNames.updatedAt].toString()) {
```

Comment:
Too long statement.

Reply:
Yep, I'll take care of that in next push